clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/EntityRefIdBuilder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.elasticsearch;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class EntityRefIdBuilder {
+  public static EntityRefId buildId(String cloudProvider, String entityType, String entityId, String account, String region) {
```

Comment:
After some back and forth ... decided to go with an explicit `id` generation scheme of `{{cloudProvider}}:{{entityType}}:{{entityId}}:{{account}}:{{region}}` instead of trying to leverage the existing cloudprovider-specific `Keys` generators.

I didn't see anything significant in the existing the existing `Keys` implementations that wasn't compatible w/ `cloudProvider:entityType:entityId:account:region`. 

`account` and `region` can be optional.